### ISSUE TITLE
Testing on multiple client request

### ISSUE THREAD

**Issue by greenyourcode** [apollographql/apollo-angular#1254]
Hello Everybody ! 

I'm looking for testing GraphQL Apollo request in case of multiple client. 

Project are based on micro service architecture so there are many client. 
In the component, i wrote : this.apollo.use('domain_1'), that's works fine ! 
[https://www.apollographql.com/docs/angular/features/multiple-clients/](url)

But my problem concerns of testing this component ! 
In my case i would like to test my request on 'domain_1' client. 

Here's the error : "TypeError: Cannot read property 'query' of undefined"
I don't know how to mock the "use()" function of Apollo. Use() is undefined !!! 

Component : 
```
`public callDummyAPI(): any {
    apollo.create({
      link: httpLink.create({ uri }),
      cache: new InMemoryCache(),
    });
    return this.apollo.use('domain_1').query<QueryType>({
      query: MY_QUERY,
    })
    .pipe(
      map(result => result.data.hello)
    );
  }`
```

Test component : 
```
`it('should test search', (done) => {
    const fixture = TestBed.createComponent(DummyComponent);
    const component = fixture.debugElement.componentInstance;
    component.callDummyAPI().subscribe(result => {
      expect(result).toEqual('tetete');
      done();
    });
    controller.expectOne('/', MY_QUERY).flush({
      data : {
        hello: 'tetete'
      }
    });
  });`
```

Does anyone knowing to resolve this problem ? 
Thank you very much in advance. 
Val

**Comment by Linkwink** [comment]
Same problem for me +1

**Comment by Linkwink** [comment]
Solution not perfect but it's works:
```
 .....
 import { ApolloTestingModule, ɵa as ApolloTestingBackend } from 'apollo-angular/testing';
 .....

 let apollo: Apollo:
 let backend: ApolloTestingBackend;

 beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        ...
      ],
      imports: [
        ...
        ApolloTestingModule,
      ],
    });

  apollo = TestBed.get(Apollo); 

  apollo.createNamed(ApolloClientName.API_V2, {
        link: new ApolloLink(operation => backend.handle(operation)),
        cache: new InMemoryCache({
          addTypename: false,
        }),
      });

  apollo.use(ApolloClientName.API_V2);
  
  });
....
```

**Comment by nbaua** [comment]
Hi,

Apologies for hijacking the thread, However there is no correct sample it seems which shows the `apollo.createNamed` scenario.
We also do have multiple micro-services and hence need to connect to multiple clients.
Also on top of that, we are using NestJS 

NestJS, allows us to inject the httplink using Factory provider, 

**Inside AppModule providers:[]**

```
useFactory: (httpLink: HttpLink) => {
                return {
                    cache: new InMemoryCache(),
                    link: httpLink.create({
                        // uri: config.servicesBaseUrl
                        uri(operation) {
                            if (!operation.operationName) {
                                return config.servicesBaseUrl;
                            }
                            return operation.operationName.toLowerCase() === 'login'
                                ? config.servicesLoginUrl
                                : config.servicesBaseUrl;
                        },
                    }),
                };
            },](url)
```

However we are not able to do so with the **ApolloLink.from** OR **apollo.createNamed** ,

It would be good if someone can suggest how to achieve this with the NestJS.

Thanks

**Comment by kamilkisiela** [comment]
#1365 will enable to test named clients

**Comment by kamilkisiela** [comment]
Released `v1.8.0` - check out ["Testing named clients"](https://www.apollographql.com/docs/angular/guides/testing/#using-named-clients) in docs
